workflow/controller/controller.go

```diff
@@ -510,6 +510,26 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 			if err != nil && !apierr.IsNotFound(err) {
 				return err
 			}
+		case remarkExitCode:
+			innerPod, err := wfc.getPod(namespace, podName)
+			if innerPod == nil || err != nil {
+				return err
+			}
+			for _, c := range innerPod.Spec.Containers {
+				if c.Name == common.WaitContainerName {
+					//exec cmd  /bin/sh -c "echo 137 > /var/run/argo/ctr/main/exitcode"
+					toPath := common.VarRunArgoPath + "/ctr/" + common.MainContainerName + "/exitcode"
+					command := []string{"/bin/sh", "-c", fmt.Sprintf("echo 137 > %s", toPath)}
```

Comment:
this cannot work as `sh` is not installed on the wait container anymore

Reply:
when main container been killed by OOM-killer(cgroup or host os), no 'exitcode' file wirte to ctr/main dir, but wait container stuck in running state,  up to this condition, just use sh command write a FAKE exitcode file, this FAKE exitode will trigger wait-container exit。

Reply:
this line can work ? 
[command := []string{"/bin/sh", "-c", "kill -%d 1"}](https://github.com/argoproj/argo-workflows/blob/2d1758fe90fd60b37d0dfccb55c3f79d8a897289/workflow/signal/signal.go#L18)

Reply:
No. That is not on master anymore

Reply:
thanks for kind reply.

Reply:
@firetaker Are you still able to reproduce on Master? if not, can you close this PR?